modules/disk-uploader/pkg/secrets/secrets_test.go

```diff
@@ -0,0 +1,110 @@
+package secrets_test
+
+import (
+	"context"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek8sclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/modules/disk-uploader/pkg/secrets"
+)
+
+var _ = Describe("VMExport", func() {
+	var (
+		kubeClient *fakek8sclient.Clientset
+	)
+
+	const (
+		namespace = "test-namespace"
+		name      = "test-vmexport"
+	)
+
+	BeforeEach(func() {
+		kubeClient = fakek8sclient.NewSimpleClientset()
+
+		os.Setenv("POD_NAME", name)
+		os.Setenv("POD_NAMESPACE", namespace)
+	})
+
+	AfterEach(func() {
+		os.Unsetenv("POD_NAME")
+		os.Unsetenv("POD_NAMESPACE")
+	})
+
+	Describe("CreateVirtualMachineExportSecret", func() {
+		It("should return no error when created", func() {
+			_, err := kubeClient.CoreV1().Pods(namespace).Create(context.Background(),
+				&corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+				},
+				metav1.CreateOptions{},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return error when set pod owner reference failed", func() {
+			err := secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError("pods \"test-vmexport\" not found"))
+		})
+	})
+
+	Describe("GetTokenFromVirtualMachineExportSecret", func() {
+		It("should return token when exists", func() {
+			kubeClient.Fake.PrependReactor("get", "secrets", func(action testing.Action) (bool, runtime.Object, error) {
```

Comment:
Create a secret with the fake client instead?

Reply:
Ah I missed it. Thanks!